### DISCUSSION TITLE
Allow feeds to use OAuth as authentication

### DISCUSSION THREAD

**Discussion by engram-design** [engram-design/FeedMe#886] [Ideas]

**Comment by janmeeus** [comment]
@engram-design Any news on the progress of the OAuth implementation?

**Comment by pieterjandebruyne** [comment]
Bump, Is it yet possible to import from google sheets API using OAuth ?

**Comment by ghost** [comment]
Any news on this?

**Comment by siebird** [comment]
I have a use case for this with Zoom's API which requires `oauth2`. I could probably hack something together with [OAuth 2.0 Client](https://github.com/venveo/craft-oauthclient) and point FeedMe to a template.

**Comment by mark-chief** [comment]
Bumpety Bump -  this would be awesome!

**Comment by pghorpade** [comment]
Any updates on this?

**Comment by maxstrebel** [comment]
I am currently in the situation in a project I took on. It relies on a timed airtable import. Airtable deprecated their api-key based API and forces oAuth ([Airtable Deprecation Notice](https://support.airtable.com/docs/airtable-api-key-deprecation-notice)). I cannot see a way to around this.

**Reply by siebird** [reply]
You could probably use the [Consume plugin](https://verbb.io/craft-plugins/consume/features) and then point feed me to that template/url.

It's one more dependency, but should work in theory.

**Reply by dmatthams** [reply]
Tokens are replaced with [ Personal Access Tokens](https://airtable.com/create/tokens) so you do not have to use oAuth. You may however need to put the key in an [Authorization header](https://docs.craftcms.com/feed-me/v4/get-started/configuration.html#example-requestoptions) rather than as a query string param. I'm going to be doing this soon also 😉

**Reply by maxstrebel** [reply]
Hey, thanks for the quick replies! 

@dmatthams how would I do that in feed-me? Do you know that? Sorry I am out of my water here. I saw that with curl it is really easy (`curl "https://api.airtable.com/..." \ -H "Authorization: Bearer PERSONAL_TOKEN"`. But I do not know how to make feed-me pass this bearer along.

@siebird, thanks a lot! If I do not find a more vanilla way, I will go that route.

**Reply by dmatthams** [reply]
Your `feed-me.php` (see [the docs](https://docs.craftcms.com/feed-me/v4/get-started/configuration.html#example-requestoptions)) will look something like:

```
<?php

return [
    '*' => [
        'feedOptions' => [
            '1' => [
                'feedUrl' => 'https://specialurl.io/feed.json',
                'requestOptions' => [
                    'headers' => [
                        'Authorization', 'Bearer YOUR_TOKEN'
                    ],
                ],
            ]
        ],
    ]
];
```
Where '1' is the ID of your feed.

**Reply by maxstrebel** [reply]
Oh, nevermind, I found a way! You can set request options in a `/config/feed-me.php` file and pass the bearer there.

```php
<?php

return [
    'feedOptions' => [
        '5' => [
            'requestOptions' => [
                    'headers' => [
                        'Authorization' => 'Bearer ' . getenv('AIRTABLE_BEARER_TOKEN'),
                        'Accept' => 'application/json',
                    ]
            ],
        ]
    ]
];
```

**Reply by maxstrebel** [reply]
Thanks so much for your help. Works great.
